test/test-applications/integrations/Samples.WebRequest.NetFramework20/Program.cs

```diff
@@ -0,0 +1,138 @@
+            // like it does in .NET Framework.
+            Environment.Exit(0);
+        }
+
+        public static HttpListener StartHttpListenerWithPortResilience(string port, int retries = 5)
+        {
+            // try up to 5 consecutive ports before giving up
+            while (true)
+            {
+                Url = $"http://localhost:{port}/Samples.WebRequest/";
+
+                // seems like we can't reuse a listener if it fails to start,
+                // so create a new listener each time we retry
+                var listener = new HttpListener();
+                listener.Prefixes.Add(Url);
+
+                try
+                {
+                    listener.Start();
+
+                    listenerThread = new Thread(HandleHttpRequests);
+                    listenerThread.Start(listener);
+
+                    return listener;
+                }
+                catch (HttpListenerException) when (retries > 0)
+                {
+                    // only catch the exception if there are retries left
+                    port = TcpPortProvider.GetOpenPort().ToString();
+                    retries--;
+                }
+
+                // always close listener if exception is thrown,
+                // whether it was caught or not
+                listener.Close();
+            }
+        }
+
+        private static void HandleHttpRequests(object state)
+        {
+            var listener = (HttpListener)state;
+
+            while (listener.IsListening)
+            {
+                try
+                {
+                    var context = listener.GetContext();
+
+                    Console.WriteLine("[HttpListener] received request");
+
+                    // read request content and headers
+                    using (var reader = new StreamReader(context.Request.InputStream, context.Request.ContentEncoding))
+                    {
+                        string requestContent = reader.ReadToEnd();
+                        Console.WriteLine($"[HttpListener] request content: {requestContent}");
+
+                        foreach (string headerName in context.Request.Headers)
+                        {
+                            string headerValue = context.Request.Headers[headerName];
+                            Console.WriteLine($"[HttpListener] request header: {headerName}={headerValue}");
+                        }
+                    }
+
+                    // write response content
+                    byte[] responseBytes = Utf8.GetBytes(ResponseContent);
+                    context.Response.ContentEncoding = Utf8;
+                    context.Response.ContentLength64 = responseBytes.Length;
+                    context.Response.OutputStream.Write(responseBytes, 0, responseBytes.Length);
+
+                    // we must close the response
+                    context.Response.Close();
+                }
+                catch (HttpListenerException)
```

Comment:
I guess we are missing a couple of catch here: https://github.com/DataDog/dd-trace-dotnet/blob/5c37e00cc0059f76c1f43c1d4e55e7dafcf9f4a2/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs#L210-L222

Reply:
Thanks for pointing out, added in c63d1c59e14a449b68bcfe38805cde8612929e5c